apps/src/p5lab/spritelab/coreLibrary.js

```diff
@@ -212,6 +216,17 @@ function atTimeEvent(inputEvent, p5Inst) {
   return [];
 }
 
+function collectDataEvent(inputEvent, p5Inst) {
+  // Only log data once per second, approximate 30 frames per second
+  if (p5Inst.World.frameCount % 30 === 0) {
+    // Call callback with no extra args
+    return [{}];
```

Comment:
🤯  how does this work? (e.g., how does returning an object within the array execute the callback, but an empty array doesn't?)

Reply:
It's an array of objects, where each element of the array corresponds to one call to the user's callback. The properties in the object would specify arguments to be passed into the callback. So `{}` means call the callback with no arguments, `{x: 4}` would mean call the callback and pass `{x: 4}` as the argument to the callback.
As a result, `[]` means don't call the callback at all, while `[{}]` means call the callback once (with no args)

Reply:
Used here: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/p5lab/spritelab/coreLibrary.js#L365

Reply:
oh, cool! much simpler than the idea i had in my head 😆